python/ray/data/grouped_dataset.py

```diff
@@ -59,6 +72,38 @@ def reduce(
             list(mapper_outputs), key, aggs, finalize=not partial_reduce
         )
 
+    @staticmethod
+    def _pruneUnusedColumns(
```

Comment:
In the future, when we have a query optimizer, this logic can be moved during planning, instead of during execution time.